couchbase/cluster_options.hxx

```diff
@@ -254,6 +255,7 @@ public:
     behavior_options::built behavior;
```

Comment:
We need to provide access to be able to update the `application_telemetry_options`

```
  [[nodiscard]] auto application_telemetry() -> application_telemetry_options&
  {
    return application_telemetry_;
  }
```